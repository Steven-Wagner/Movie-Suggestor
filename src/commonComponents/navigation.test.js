import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import ReactDOM from 'react-dom';
import Nav from './navigation';

describe('<Nav/>', () => {
    it('renders without crashing', () => {
    const div = document.createElement('div');

    ReactDOM.render(
        <Router>
            <Nav user="testuser" status="homepage"/>
        </Router>,
        div);
    ReactDOM.unmountComponentAtNode(div);
    });

    describe(`renders correctly for all pages`, () => {

        const pages = ["landing-page", "homepage", "friend-suggestions-page", "review-page", "past-reviews", "signUp", "login"]

        pages.forEach(status => {

            it(`renders ${status} Nav UI as expected`, () => {
                const tree = renderer
                .create(
                    <Router>
                        <Nav user="testuser" status={status}/>
                    </Router>)
                .toJSON();
                expect(tree).toMatchSnapshot();  
            });
        })
    });
});