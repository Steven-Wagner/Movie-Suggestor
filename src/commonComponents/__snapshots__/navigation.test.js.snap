// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Nav/> renders correctly for all pages renders friend-suggestions-page Nav UI as expected 1`] = `
<div
  className="nav-wrapper"
>
  <nav>
    <a
      href="/homepage/testuser"
      onClick={[Function]}
    >
      <button
        className="nav-button remote-button"
      >
        Home
      </button>
    </a>
    <a
      href="/newreview/testuser"
      onClick={[Function]}
    >
      <button
        className="nav-button remote-button"
      >
        Review Movie
      </button>
    </a>
    <a
      href="/"
      onClick={[Function]}
    >
      <button
        className="nav-button remote-button"
        onClick={[Function]}
      >
        Logout
      </button>
    </a>
  </nav>
</div>
`;

exports[`<Nav/> renders correctly for all pages renders homepage Nav UI as expected 1`] = `
<div
  className="nav-wrapper"
>
  <nav>
    <a
      href="/friendsuggestions/testuser"
      onClick={[Function]}
    >
      <button
        className="nav-button remote-button"
      >
        Friend Suggestions
      </button>
    </a>
    <a
      href="/newreview/testuser"
      onClick={[Function]}
    >
      <button
        className="nav-button remote-button"
      >
        Review Movie
      </button>
    </a>
    <a
      href="/"
      onClick={[Function]}
    >
      <button
        className="nav-button remote-button"
        onClick={[Function]}
      >
        Logout
      </button>
    </a>
  </nav>
</div>
`;

exports[`<Nav/> renders correctly for all pages renders landing-page Nav UI as expected 1`] = `
<div
  className="nav-wrapper"
>
  <nav>
    <a
      href="/signup"
      onClick={[Function]}
    >
      <button
        className="nav-button remote-button"
      >
        Sign-up
      </button>
    </a>
    <a
      href="/login"
      onClick={[Function]}
    >
      <button
        className="nav-button remote-button"
      >
        Login
      </button>
    </a>
  </nav>
</div>
`;

exports[`<Nav/> renders correctly for all pages renders login Nav UI as expected 1`] = `
<div
  className="nav-wrapper"
>
  <nav>
    <a
      href="/signup"
      onClick={[Function]}
    >
      <button
        className="nav-button remote-button"
      >
        Sign-up
      </button>
    </a>
  </nav>
</div>
`;

exports[`<Nav/> renders correctly for all pages renders past-reviews Nav UI as expected 1`] = `
<div
  className="nav-wrapper"
>
  <nav>
    <a
      href="/homepage/testuser"
      onClick={[Function]}
    >
      <button
        className="nav-button remote-button"
      >
        Home
      </button>
    </a>
    <a
      href="/friendsuggestions/testuser"
      onClick={[Function]}
    >
      <button
        className="nav-button remote-button"
      >
        Friend Suggestions
      </button>
    </a>
    <a
      href="/newreview/testuser"
      onClick={[Function]}
    >
      <button
        className="nav-button remote-button"
      >
        Review Movie
      </button>
    </a>
    <a
      href="/"
      onClick={[Function]}
    >
      <button
        className="nav-button remote-button"
        onClick={[Function]}
      >
        Logout
      </button>
    </a>
  </nav>
</div>
`;

exports[`<Nav/> renders correctly for all pages renders review-page Nav UI as expected 1`] = `
<div
  className="nav-wrapper"
>
  <nav>
    <a
      href="/homepage/testuser"
      onClick={[Function]}
    >
      <button
        className="nav-button remote-button"
      >
        Home
      </button>
    </a>
    <a
      href="/friendsuggestions/testuser"
      onClick={[Function]}
    >
      <button
        className="nav-button remote-button"
      >
        Friend Suggestions
      </button>
    </a>
    <a
      href="/"
      onClick={[Function]}
    >
      <button
        className="nav-button remote-button"
        onClick={[Function]}
      >
        Logout
      </button>
    </a>
  </nav>
</div>
`;

exports[`<Nav/> renders correctly for all pages renders signUp Nav UI as expected 1`] = `
<div
  className="nav-wrapper"
>
  <nav>
    <a
      href="/login"
      onClick={[Function]}
    >
      <button
        className="nav-button remote-button"
      >
        Login
      </button>
    </a>
  </nav>
</div>
`;
